flink-table/flink-table-common/src/main/java/org/apache/flink/table/plan/stats/TableStats.java

```diff
@@ -69,4 +70,18 @@ public TableStats copy() {
 		return copy;
 	}
 
+	public TableStats merge(TableStats other) {
+		Map<String, ColumnStats> colStats = new HashMap<>();
+		for (Map.Entry<String, ColumnStats> entry : this.colStats.entrySet()) {
+			String col = entry.getKey();
+			ColumnStats stats = entry.getValue();
+			ColumnStats otherStats = other.colStats.get(col);
+			if (otherStats != null) {
+				colStats.put(col, stats.merge(otherStats));
+			}
+		}
+		return new TableStats(
+				Stream.of(this.rowCount, other.rowCount).anyMatch(c -> c == UNKNOWN.rowCount) ?
```

Comment:
i think the logic will be more robustness , like `this.rowCount >=0 && other.rowCount >= 0 ? ...` not `c == UNKNOWN.rowCount` (it's a public interface, other negative number may also be given by user)